### PR TITLE
#20 Configure detector 

### PR DESCRIPTION
**Pull request by punksta**
- BarcodeDetector is singleton by default
- May change it in Builder

added button to start new activity in example for singleton testing

### CODE REVIEW

File: library/src/main/java/github/nisrulz/qreader/QREader.java [hunk 1]
```diff
@@ -77,14 +80,24 @@
...
     this.autoFocusEnabled = builder.autofocusEnabled;
     this.width = builder.width;
     this.height = builder.height;
     this.facing = builder.facing;
     this.qrDataListener = builder.qrDataListener;
     this.context = builder.context;
     this.surfaceView = builder.surfaceView;
+    this.barcodeDetector = builder.barcodeDetector;
+    //for better performance we should use one detector for all Reader, if builder not specify it
+    this.barcodeDetector = builder.barcodeDetector;
```
Comment (nisrulz) [inline library/src/main/java/github/nisrulz/qreader/QREader.java]: Duplicate code. Sameas on line 96

File: app/src/main/java/github/nisrulz/projectqreader/MainActivity.java [hunk 2]
```diff
@@ -60,6 +61,13 @@ private void initAndStartQrReader(int previewWidth, int previewHeight) {
         .width(previewWidth)
         .build();
     qrEader.init();
+
+    findViewById(R.id.new_activity).setOnClickListener(new View.OnClickListener() {
+      @Override public void onClick(View view) {
+        startActivity(new Intent(MainActivity.this, MainActivity.class));
```
Comment (nisrulz) [inline app/src/main/java/github/nisrulz/projectqreader/MainActivity.java]: Why do we need this for ?

Comment (punksta) [inline app/src/main/java/github/nisrulz/projectqreader/MainActivity.java]: It shows that one BarcodeDetector works in many readers.

File: library/src/main/java/github/nisrulz/qreader/QREader.java (hunk 1 above)
Comment (punksta) [inline library/src/main/java/github/nisrulz/qreader/QREader.java]: fixed

**Comment by nisrulz** [comment]
@punksta there are a couple of things I would like to mention. So my idea was that there will be a Singleton of `QREader` class i.e you make a call like this `QREader.getInstance().init()` .

I version looks a bit different as you are making the `BarcodeDetector` as a singleton. Now think of this having a singleton `QREader` class also takes care of the `CameraSource` being referenced as a single object, which is desired. 

Also the example , I would prefer we show that in a different activity and not just restart the same activity. Just for the fact usually new devs wont understand this behaviour , but looking at a new activity being spawned and there getting the instance is pretty  straightforward.

Let me know what do you think about this.

**Comment by punksta** [comment]
@Nisrulz I agree with second point. Two activities will be more helpfull.

But disagree with first one:
- Do you really want to make control of view changing inside `QREader` and reconfigure `CameraSource` it after change?
- Are you sure, that you want to have only one state of configuration (hight, width, autofocus...) in a project?
-  The main cause of my pl is resource savings by reusing detector.

I imagine that `RQreader` acts as presenter between `BarcodeDetector`(model) and `SurfaceView`` inside `Activity`(both are View). Isn't it? I think that having self presenter on each view and singleton model is a common practice.

**Pull request merged**